src/User/src/Service/UserService.php

```diff
@@ -134,6 +135,16 @@ public function updateUser(User $user, array $data = []): User
 
         if (isset($data['isDeleted'])) {
             $user->setIsDeleted((bool) $data['isDeleted']);
+
+            if ((bool) $data['isDeleted'] === true) {
+                // make user anonymous
+                $user->setIdentity('anonymous' . date('dmYHis') . $this->config['userAnonymizeAppend']);
```

Comment:
```suggestion
                $user->setIdentity(
                    sprintf('anonymous%s@%s', date('dmYHis'), $this->config['userAnonymizeAppend'])
                );
```
You had a missing `@` between the date and the configured append part.
Also, for better readability, I suggest using `sprintf`.

Reply:
i just assumed the config (userAnonymizeAppend) would have the '@'
either way, your solution is better

Reply:
When I tested locally, I forgot that @arhimede's [comment](https://github.com/dotkernel/frontend/issues/479#issuecomment-2331111238) included both the `@` symbol and the domain name.
I just set `userAnonymizeAppend` to `dotkernel.com` and after testing, I ended up with an anonymized email without the `@` in it.
Your solution was just as correct as mine, but IMHO the `sprintf` solution looks more readable than concatenating four strings.